src/live-programming.rst

```diff
@@ -26,6 +26,64 @@ Interactive Debuggers
+
+There are **program visualization** systems designed to aid significantly the practice of professional programming and software engineering as well as the learning of computer science by novice programmers. And it turns out that **program visualization** doesn't have to deal with toy programs, e.g., `Frappe - Visualizing Code Dependency Graphs <https://softvis.wordpress.com/2018/08/31/frappe-visualizing-code-dependency-graphs/>`_.
+
+.. container:: bib-item
+
+  .. bibliography:: live-programming.bib
+    :filter: key == "price1993principled"
+
+ Note that **program visualization** is different from **software visualization** in that the latter includes the visualization of a high-level description of algorithm or a piece of software which is in contrast to **program visualization** where actual implemented code is visualized.
+
+.. container:: bib-item
+
+  .. bibliography:: live-programming.bib
+    :filter: key == "maletic2002task"
+
+ This paper defines five dimensions, Tasks, Audience, Target, Representation and Medium to describe several software visualization systems that have very different features along those defined dimensions. By realigning taxonomies with the perspective of current software engineering problems, open research issues are identified. Additionally, the paper identifies the strengths of individual tools and the techniques they apply.
+
+.. container:: bib-item
+
+  .. bibliography:: live-programming.bib
+    :filter: key == "sorva2013review"
+
+ This paper serves as a survey of program visualization systems whose **task** is to aid the learning and teaching of introductory programming, with an intended **audience** of novice programmers and *CS1* teachers. Even a visualization that has been painstakingly crafted to be as lucid as possible may fail to aid learning in practice. Therefore, the paper introduces a new taxonomy to describe how program visualization systems engage learners.
+
+.. container:: bib-item
+
+  .. bibliography:: live-programming.bib
+    :filter: key == "murphy2010interactive"
+
+ **Code smells** are characteristics of software that indicate that code may have a design problem. The paper propose a novel smell detector, Stench Blossom, that provides an iteractive ambient visualization designed to first give programmers a quick, high-level overview of the smells, and then to help in understanding the smells if users wish. As a result, the experiment confirmed that programmers identify more smells and make more confident and informed refactoring jedgements using the tool than not using the tool.
+
+.. container:: bib-item
+
+  .. bibliography:: live-programming.bib
+    :filter: key == "reiss2005paradox"
+
+ This paper argues that most past and current work in the field is out of touch with the reality of software development and that new approaches and new ideas are needed.
```

Comment:
Hmm, is there some more specific insight that this paper has? This is very vague.